app/src/services/xdai.ts

```diff
@@ -135,15 +192,42 @@ class XdaiService {
     }
   }
 
-  claim = async (messages: string[], signatures: string[]) => {
+  claim = async (addresses: string[], messages: string[], signatures: string[]) => {
     const multiclaim = new ethers.Contract(MULTI_CLAIM_ADDRESS, multiClaimAbi, this.provider.signer.signer)
-    return multiclaim.claim(messages, signatures)
+
+    return multiclaim.claim(addresses, messages, signatures)
   }
 
   static encodeRelayTokens = (receiver: string): string => {
     const transferFromInterface = new utils.Interface(xdaiBridgeAbi)
     return transferFromInterface.functions.relayTokens.encode([receiver])
   }
+  static encodeApprove = (address: any, value: any) => {
+    const transferFromInterface = new utils.Interface(approveAbi)
+    return transferFromInterface.functions.approve.encode([address, value])
+  }
+  static encodeTokenBridgeTransfer = (receiver: string, amount: BigNumber, data: any) => {
+    const transferFromInterface = new utils.Interface(omenAbi)
+    return transferFromInterface.functions.transferAndCall.encode([receiver, amount, data])
+  }
+
+  static waitForBridgeMessageStatus = async (hash: string, provider: any) => {
+    let status
+    while (!status) {
+      // get tx receipt from mainnet
+      const transaction = await provider.signer.signer.provider.getTransactionReceipt(hash)
+      if (transaction) {
+        // get messageId from transaction logs
+        const log = transaction.logs[transaction.logs.length - 1]
+        const messageId = log.topics[log.topics.length - 1]
+        // check if the message has been processed
+        const xdaiOmniAddress = '0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59'
```

Comment:
This address should be a constant right?

Reply:
Dunno how that slipped! Good catch